### ISSUE TITLE
index auto-name collisions

### ISSUE THREAD

**Issue by qwesda** [jakob/Postico#539]
## What did you do?

If you take this table 

```sql
CREATE TABLE "public"."test" (
    "id" serial,
    "flag1" bool NOT NULL DEFAULT 'false',
    "flag2" bool NOT NULL DEFAULT 'false',
    PRIMARY KEY ("id")
);
```

and use the structure view to create two expression indices like `flag1 = true` and `flag2 = true`. The auto generated names for the indices are both `test_expr_idx`.

## What did you expect to happen?

The index names should be checked for collision with existing indices and the other ones that will be created. 

Or the index name should not be specified in the create query (if not explicitly specified by the user)  so that postgres takes care of naming them.

## What actually happened?

Save changes fails with `PostgreSQL said: relation "test_expr_idx" already exists`;

## What software versions are you using?

Postico version: 1.3.5
macOS version: 10.13.4
PostgreSQL version: 10.3

**Comment by jakob** [comment]
Good catch. The best course of action would probably be to leave the name generation up to the PostgreSQL. If I remember correctly, the reason why Postico tries to generate a name is only for compatibility with older versions. Postico could check the server version, and generate an index name only if necessary.

**Comment by jakob** [comment]
Also since we have a bug bounty, please send me an email to jakob@eggerapps.at and let me know what name/email I should put on your free license!
